mypy/plugins/attrs.py

```diff
@@ -505,6 +505,8 @@ def _attribute_from_attrib_maker(ctx: 'mypy.plugin.ClassDefContext',
     # If the type isn't set through annotation but is passed through `type=` use that.
     type_arg = _get_argument(rvalue, 'type')
     if type_arg and not init_type:
+        if ctx.api.options.python_version >= (3, 6):
+            ctx.api.note('Variable annotation syntax is preferred over type= arguments', rvalue)
```

Comment:
I think that this too noisy to be enabled by default. Some users might want to type check their codebase using multiple Python versions, and this could result in many notes when using 3.6 or later as the target Python version. I think that this would work better as an optional check in the future, once we support enabling and disabling individual checks (I can't give a timeline for this feature though).

Reply:
I tend to agree, however I would still appreciate this included in nearish future. How about if the note would be emitted just once for the first occurrence in a run, and message changed to note the Python version, something like:

> Variable annotation syntax is preferred over type= arguments with Python 3.6 and later

Do you think that would be acceptable already now?

Reply:
Any `note` makes mypy exit with a 1 exit code, so I don't think it's realistically possible to have this note at all.